src/test/java/pl/maciejwalkowiak/drools/BeforeMethodBasedTest.java

```diff
@@ -22,9 +23,10 @@ public void initDrools() throws Exception {
     public void should_set_discount() {
         Purchase purchase = new Purchase(new Customer(17));
 
-        session.insert(purchase);
-        session.fireAllRules();
+        session.execute(purchase);
 
+        System.out.println("isDiscount?: " + purchase.getTicket().isDiscount());
```

Comment:
There is no need to print out anything in unit tests, and if there is for some reason - use logger.